src/document-state.js

```diff
@@ -112,6 +116,31 @@ export class DocumentState {
   onVisibilityChanged_() {
     this.visibilityObservable_.fire();
   }
+
+  /**
+   * @param {function()} handler
+   * @return {!UnlistenDef|null}
+   */
+  onBodyAvailable(handler) {
+    const doc = this.document_;
+    if (!!doc.body) {
+      handler();
+      return null;
+    }
+    if (!this.bodyAvailableObservable_) {
+      this.bodyAvailableObservable_ = new Observable();
+      waitForChild(doc.documentElement,
+                   () => !!doc.body,
+                   this.onBodyAvailable_.bind(this));
+    }
+    return this.bodyAvailableObservable_.add(handler);
+  }
+
+  /** @private */
+  onBodyAvailable_() {
+    this.bodyAvailableObservable_.fire();
+    this.bodyAvailableObservable_ = null;
```

Comment:
We're going to have a memory leak here if anybody stores the `unlisten` returned by `#onBodyAvailable`. We'll need to add a `Observable#clear`.